odoo/addons/test_inherits/tests/test_inherits.py

```diff
@@ -138,3 +138,18 @@ def test_display_name(self):
         self.assertIn('lang', Unit.display_name.depends_context)
         self.assertIn('lang', Box.display_name.depends_context)
         self.assertIn('lang', Pallet.display_name.depends_context)
+
+    def test_multi_write_m2o_inherits(self):
+        """Verify that an inherits m2o field can be written to in batch"""
+        Unit = self.env['test.unit']
+        unit_foo = Unit.create({'name': 'foo'})
+        unit_bar = Unit.create({'name': 'bar'})
+
+        Box = self.env['test.box']
+        boxes = Box.browse([])
+
+        for i in range(5):
+            boxes |= Box.create({'unit_id': unit_foo.id})
+
+        self.assertTrue(boxes.write({'unit_id': unit_bar.id}))
+        self.assertEquals(boxes.mapped('unit_id.name'), ['bar'])
```

Comment:
It could be simpler:
```suggestion
        unit_foo, unit_bar = self.env['test.unit'].create([{'name': 'foo'}, {'name': 'bar'}])
        boxes = self.env['test.box'].browse([{'unit_id': unit_foo.id} for i in range(5)])
        boxes.unit_id = unit_bar
        self.assertEquals(boxes.mapped('unit_id.name'), ['bar'])
```